src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/builder/HaFlowBuilder.groovy

```diff
@@ -163,28 +159,10 @@ class HaFlowBuilder {
         return this
     }
 
-    /**
-     * Adds ha-flow and waits for it to become UP.
-     */
-    HaFlowExtended create() {
-        log.debug("Adding ha-flow")
-        def response = northboundV2.addHaFlow(haFlowRequest)
-        assert response.haFlowId
-        HaFlow haFlow
-        Wrappers.wait(FLOW_CRUD_TIMEOUT) {
-            haFlow = northboundV2.getHaFlow(response.haFlowId)
-            assert haFlow.status == FlowState.UP.toString()
-                    && haFlow.getSubFlows().status.unique() == [FlowState.UP.toString()], "Flow: ${haFlow}"
-        }
-        new HaFlowExtended(haFlow, northboundV2, topologyDefinition)
-    }
-
-    /**
-     * Adds ha-flow without waiting for successful creation
-     */
-    HaFlow add() {
-        log.debug("Send request to create ha-flow")
-        northboundV2.addHaFlow(haFlowRequest)
+    HaFlowExtended build() {
+        log.debug("Adding HA-Flow")
+        def haFlow = northboundV2.addHaFlow(haFlowRequest)
```

Comment:
Hm. I'd implement it the way
```
def haFlow = new HaFlowExtended(haFlowRequest, northboundV2, topologyDefinition)
haFlow.create() //Here we can also wait for flow to go up.
return haFlow
```

Reply:
Decided to split the logic of adding an entity to the system and successful creation with status verification.  The method getRandom() from HaFlowFactory includes this verification.